### PR TITLE
Add two controllers (todolists and records)

### PR DESCRIPTION
**Pull request by Skr1pt1k**
Add two controllers

### CODE REVIEW

File: app/controllers/records_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    task = Record.find(params[:id])
+    task.update_attributes(record_params)
+    render json: task
+  end
+
+  def destroy
+    task = Record.find(params[:id])
+    task.destroy
+    head :no_content, status: :ok
+    end
```
Comment (alex-pro) [inline app/controllers/records_controller.rb]: fix tabulation

File: app/controllers/records_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  private
+  def set_record
+    @record = Record.find(params[:id])
```
Comment (alex-pro) [inline app/controllers/records_controller.rb]: use ||=

File: app/controllers/todo_lists_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+class TodoListsController < ApplicationController
+  def index
+    todos = TodoList.all
```
Comment (alex-pro) [inline app/controllers/todo_lists_controller.rb]: inapropriate var name

File: app/controllers/todo_lists_controller.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    render json: todos
+  end
+
+  def create
+    todo = TodoList.create(todo_param)
```
Comment (alex-pro) [inline app/controllers/todo_lists_controller.rb]: same new then save way should be here

File: config/database.yml [hunk 5]
```diff
@@ -24,6 +24,9 @@ default: &default
 development:
```
Comment (alex-pro) [inline config/database.yml]: put this file to gitignore. instead, commit database.yml.example file which will be used like a template by the new devs

File: config/routes.rb [hunk 6]
```diff
@@ -1,3 +1,9 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get 'records/index'
```
Comment (alex-pro) [inline config/routes.rb]: use resources instead of get

File: db/migrate/20181211080206_create_records.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+class CreateRecords < ActiveRecord::Migration[5.2]
+  def change
+    create_table :records do |t|
+      t.string :title
+      t.boolean :done
+      t.date :date
+      t.string :details
+      t.integer :priority
+
```
Comment (alex-pro) [inline db/migrate/20181211080206_create_records.rb]: here should be todolist refference

File: app/controllers/records_controller.rb [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class RecordsController < ApplicationController
+  before_action :set_record, only: %i[update destroy]
+  def index; end
+
+  def create
+    @task = Record.create(record_params)
```
Comment (alex-pro) [inline app/controllers/records_controller.rb]: create has two main methods inside: initialize and save. so your next line save record which is already saved in db. rework to var = Model.new(params); if var.save then...

File: app/controllers/records_controller.rb [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    if @task.save
```
Comment (alex-pro) [inline app/controllers/records_controller.rb]: moreover, there is no need to use instance variables. for API use simple variable (without @)

File: app/controllers/records_controller.rb [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+      render json: @task, message: 'created', status: :created
+    else
+      render json: @task.errors, message: 'error', status: :unprocessable_entity
+    end
+  end
+
+  def update
+    @record.update(record_params)
```
Comment (alex-pro) [inline app/controllers/records_controller.rb]: use condition update :
if update then do smth else do smth
in purpose to handle errors

File: app/controllers/records_controller.rb [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    render json: task, message: 'updated', status: :ok
+  end
+
+  def destroy
+    task = Record.find(params[:id])
```
Comment (alex-pro) [inline app/controllers/records_controller.rb]: set_record is already do the same. duplication

File: app/controllers/todo_lists_controller.rb [hunk 12]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class TodoListsController < ApplicationController
+  def index
+    todo = TodoList.all
```
Comment (alex-pro) [inline app/controllers/todo_lists_controller.rb]: rename to todo_lists so it'll become appropriate

File: app/controllers/todo_lists_controller.rb [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+
+  def destroy
+    todo = TodoList.find(params[:id])
+    todo.destroy
+    head :no_content, status: :ok
+  end
+
+  private
+
+  def todo_param
```
Comment (alex-pro) [inline app/controllers/todo_lists_controller.rb]: should be todo_list_params

File: bin/bundle [hunk 14]
```diff
@@ -1,3 +1,5 @@
 #!/usr/bin/env ruby
```
Comment (alex-pro) [inline bin/bundle]: you can remove all bin init files from rubocop inspection

File: client/src/component/todoLists.js.coffee [hunk 15]
```diff
@@ -0,0 +1,7 @@
+@TodoLists = React.createClass
```
Comment (alex-pro) [inline client/src/component/todoLists.js.coffee]: do not forget to remove it. in future the project will be parted into two repo

File: config/routes.rb [hunk 16]
```diff
@@ -1,3 +1,8 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :records
+  scope '/api/v1' do
```
Comment (alex-pro) [inline config/routes.rb]: it is not necessary to use api namespaces for our case

File: db/migrate/20181211080206_create_records.rb [hunk 17]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class CreateRecords < ActiveRecord::Migration[5.2]
+  def change
+    create_table :records do |t|
+      t.references :todo_list
```
Comment (alex-pro) [inline db/migrate/20181211080206_create_records.rb]: add index for reference

File: db/schema.rb [hunk 18]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (alex-pro) [inline db/schema.rb]: you can add schema.rb file to rubocop ignore list

File: db/seeds.rb [hunk 19]
```diff
@@ -10,5 +10,5 @@
 # TodoList.create(title: "First toDoList")
 # TodoList.create(title: "Second toDoList")
 # TodoList.create(title: "X toDoList")
-Record.create(todo_list_id: 1, title: "clean the house", done: false, date: "09/14/2009", details: "none", priority: 1)
-puts "data loaded success"
\ No newline at end of file
+Record.create(todo_list_id: 1, title: 'clean the house', done: false, date: '09/14/2009', details: 'none', priority: 1)
```
Comment (alex-pro) [inline db/seeds.rb]: better to associate todo with real record of todolist

File: app/controllers/todo_lists_controller.rb [hunk 20]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class TodoListsController < ApplicationController
+  def index
+    todo_lists = TodoList.all
+    render json: todo_lists
```
Comment (alex-pro) [inline app/controllers/todo_lists_controller.rb]: add status ok

File: app/controllers/todo_lists_controller.rb [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+    todo = TodoList.new(todo_list_params)
+    if todo.save
+      render json: todo, message: 'created', status: :created
+    else
+      render json: todo.errors, message: 'error', status: :unprocessable_entity
+    end
+  end
+
+  def update
+    todo = TodoList.find(params[:id])
```
Comment (alex-pro) [inline app/controllers/todo_lists_controller.rb]: move it to before action

File: app/controllers/todo_lists_controller.rb [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+    todo.update(todo_list_params)
```
Comment (alex-pro) [inline app/controllers/todo_lists_controller.rb]: remove this line

File: .gitignore [hunk 23]
```diff
@@ -21,3 +21,8 @@
 
 # Ignore master key for decrypting credentials and more.
 /config/master.key
+
+# My ignore files
+.idea/
+bin/*
```
Comment (alex-pro) [inline .gitignore]: do not ignore bin files

File: app/controllers/records_controller.rb [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+  expose :record
+
+  def index
+    render json: records, status: :ok
+  end
+
+  def create
+    task = Record.new(record_params)
+    if task.save
+      render json: task, message: 'created', status: :created
```
Comment (alex-pro) [inline app/controllers/records_controller.rb]: message will be overwritten so no need to put it in code outside of response body hash

Comment (Skr1pt1k) [reply app/controllers/records_controller.rb]: ok, will be done soon :)

File: app/controllers/todo_lists_controller.rb [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+    else
+      render json: todo_list.errors, message: 'error', status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    if todo_list.destroy
+      head :no_content, status: :ok
+    else
+      render json: todo_list.errors, status: :unprocessable_entity
```
Comment (alex-pro) [inline app/controllers/todo_lists_controller.rb]: you can easily return head :no_content, status: :ok in all cases. if something happens you anyway won't be able to throw todo_list.errors

File: app/models/todo_list.rb [hunk 26]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class TodoList < ApplicationRecord
+  has_many :records
```
Comment (alex-pro) [inline app/models/todo_list.rb]: add dependent: :destroy

File: config/routes.rb [hunk 27]
```diff
@@ -1,3 +1,6 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :todo_lists, except: %i[new edit show]
+  resources :records, only: %i[index create update destroy]
```
Comment (alex-pro) [inline config/routes.rb]: you can use except attribute to make the route shorter (like a row above). Anyway, choose except or only

File: db/seeds.rb [hunk 28]
```diff
@@ -9,4 +9,14 @@
...
 # TodoList.create(title: "First toDoList")
 # TodoList.create(title: "Second toDoList")
-# TodoList.create(title: "X toDoList")
\ No newline at end of file
+# TodoList.create(title: "X toDoList")
+#
+#
+todo = TodoList.create(title: 'X toDoList')
+
+Record.create(todo_list_id: todo[:id],
```
Comment (alex-pro) [inline db/seeds.rb]: you can use association instead: todo_list: todo_list
also use faker and add several records

File: app/controllers/records_controller.rb [hunk 29]
```diff
@@ -34,6 +34,10 @@ def destroy
 
   private
 
+  def serialize(param)
```
Comment (alex-pro) [inline app/controllers/records_controller.rb]: serialize what? change method name so it is easy to understand what it does
also change param to params

File: db/seeds.rb [hunk 30]
```diff
@@ -10,13 +10,20 @@
...
+todo_list = TodoList.create(title: 'X toDoList')
+
+Record.create(todo_list: todo_list,
               title: 'clean the house',
               done: false,
               date: '11-11-2019',
               details: 'fast do it!',
               priority: 4)
+
+15.times do
```
Comment (alex-pro) [inline db/seeds.rb]: you can use faker to populate record columns as well 
and improve it to have 5 records on 1 todo_list so total count should be 10 (2 todo_list)

File: app/controllers/todo_lists_controller.rb [hunk 31]
```diff
@@ -0,0 +1,48 @@
...
+  def index
+    render json: todo_list_serialize(todo_lists), status: :ok
+  end
+
+  def create
+    todo = TodoList.new(todo_list_params)
+    if todo.save
+      render json: todo_list_serialize(todo), message: 'created', status: :created
+    else
+
```
Comment (alex-pro) [inline app/controllers/todo_lists_controller.rb]: remove empty line

**Pull request merged**